vslib/src/SwitchStateBaseMACsec.cpp

```diff
@@ -748,40 +748,14 @@ sai_status_t SwitchStateBase::loadMACsecAttrsFromACLEntry(
     return SAI_STATUS_NOT_IMPLEMENTED;
 }
 
-sai_status_t SwitchStateBase::getMACsecAttr(
-        _In_ const std::string &serializedObjectId,
-        _In_ uint32_t attrCount,
-        _Out_ sai_attribute_t *attrList)
-{
-    SWSS_LOG_ENTER();
-
-    for (uint32_t i = 0; i < attrCount; i++)
-    {
-        if (attrList[i].id == SAI_MACSEC_ATTR_SCI_IN_INGRESS_MACSEC_ACL)
-        {
-            attrList[i].value.booldata = true;
-        }
-        else
-        {
-            auto meta = sai_metadata_get_attr_metadata(SAI_OBJECT_TYPE_MACSEC, attrList[i].id);
-
-            SWSS_LOG_WARN("Cannot get attribute %s", meta->attridname);
-
-            return SAI_STATUS_NOT_IMPLEMENTED;
-        }
-    }
-
-    return SAI_STATUS_SUCCESS;
-}
-
 sai_status_t SwitchStateBase::getMACsecSAAttr(
         _In_ const std::string &serializedObjectId,
         _In_ uint32_t attrCount,
         _Out_ sai_attribute_t *attrList)
 {
     SWSS_LOG_ENTER();
 
-    CHECK_STATUS(get_internal(SAI_OBJECT_TYPE_MACSEC_SA, serializedObjectId, attrCount, attrList));
+    CHECK_STATUS(get(SAI_OBJECT_TYPE_MACSEC_SA, serializedObjectId, attrCount, attrList));
```

Comment:
this and below  loop seems wrong to me, get() api will already populate all attributeList, and iy seems like you in 765 doing again some population? if those attributes must be refreshed, then they need to be done in refresh functions,
basically there should be no custom GET api implementations

Reply:
Actually, this function will never be used currently. We haven't had a finally conclusion about how to implement fetching the packet number of MACsec SA. Guohan's suggestion is we can skip this feature temporarily.
But I will remove this function if it's not acceptable to you.

Reply:
i think you can leave it if you want, ideally we would not like to have dead code in source, but if you gonna leave it, add TODO comment, with statement that i made, that get should have not other custom attribute changes

Reply:
I removed it and backup this version in my private repo. Please check it.